app/routes.rb

```diff
@@ -35,32 +20,38 @@ def with_json_body
 end
 
 get '/guides/:id/raw' do
-  guides.find(id: params['id']).projection(_id: 0).to_a.first.to_json
+  GuideCollection.find(params['id']).raw.to_json
 end
 
 get '/guides/:id' do
-  guides.find(id: params['id']).projection(_id: 0).map {|it| GitIo::Guide.new(it) }.to_a.first.to_json
+  GuideCollection.find(params['id']).to_json
+end
+
+get '/guides/:organization/:repository/raw' do
+  GuideCollection.find_by_slug(params['organization'], params['repository']).raw.to_json
+end
+
+get '/guides/:organization/:repository' do
+  GuideCollection.find_by_slug(params['organization'], params['repository']).to_json
 end
 
 post '/guides' do
   with_json_body do |body|
-    id = {id: new_id}
-    guides.insert_one body.merge(id)
-    id.to_json
+    GuideCollection.insert(body).to_json
+  end
+end
+
+put '/guides/:id' do
+  with_json_body do |body|
+    GuideCollection.update(params[:id], body).to_json
   end
 end
 
 post '/guides/import/:organization/:name' do
-  repo = GitIo::Repo.new(params[:name], params[:organization])
+  repo = GitIo::Repo.new(params[:organization], params[:name])
   guide = GitIo::Operation::Import.new(GitIo::Bot.from_env, repo, guides).run!
 
   guide.to_json
```

Comment:
We could remove all the `to_json` and [do the conversion in an after block](http://stackoverflow.com/a/27654549/1188303):

```ruby
after do
  response.body = JSON.dump(response.body)
end
```